packages/client/lib/execution/vmexecution.ts

```diff
@@ -48,6 +54,10 @@ export class VMExecution extends Execution {
 
       const stateManager = new DefaultStateManager({
         trie,
+        cacheOptions: {
+          deactivate: false,
+          size: 2500000,
```

Comment:
This immediately allocates these items, do we want this?

Reply:
Hmm, yeah, maybe this can get optimized, I would also say that it doesn't hurt too much right now (and I am also not so motivated to improve here atm).

Are you generally sure about this behavior? Where did you "see" that?

Reply:
![image](https://user-images.githubusercontent.com/29359032/228769518-24d10e87-9c45-4e24-b67c-704a38596f5d.png)

It's in the docs :) https://www.npmjs.com/package/lru-cache

Reply:
Quick note: if you set the cache size to `1` then the tests all fail (this implies that somewhere we use the cache wrong). If you use the current setting, the tests will go out-of-memory.

Reply:
(at least on the version I tested yesterday)

Reply:
Ah, that's really interesting, even 10, 100, and 1000 fail for `npm run tape -- test/execution/vmexecution.spec.ts`, 10000 is the first setting where this goes through.

Will have a look.

Reply:
Ah, one thing I am side-realizing while debugging: when we are running the client tests on mainnet, there is always (on every test run) the need to re-fill the state with all these hundres of  genesis/presale accounts, this will very likely slow down test runs significantly. So we might want to switch over to sepolia or so where ever we can.

Reply:
Update: ah, so these numbers from above, succeed or not, is likely the number of presale wallets.

Reply:
Ok, yeah, it is that the current design is just not working properly if the cache size is lower than number of accounts modified during a block execution.

This is since these modifications are only put into the trie on `flush()` execution, and if there are more puts than the cache can hold in between the old (not yet saved) values are getting out of the cache again before this is persisted.

This would need a substantial cache re-write (again) if this should be tackled, I wouldn't want to do that. But we should definitely point this out in the docs.

Reply:
[49df9605133642d40b456505188ea2edad7002ab](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2592/commits/b6eafac9cdd62637a3f7274c924d63d4c75d11d6)